File: BookClub96/BookClub96/Data/Entities/Group.cs
using System;
using System.Collections.Generic;

namespace BookClub96.Data.Entities
{
    public enum GroupType
    {
        Open, 
        Closed,
        ByApplication
    }

    public class Group
    {
        public int Id { get; set; }
        public DateTime CreationTime { get; set; }
        public ICollection<Meeting> Meetings { get; set; }
        public GroupType Type { get; set; }
        public string GroupName { get; set; }
        public ICollection<GroupMember> Members { get; set; }
    }
}

File: BookClub96/BookClub96/ClientApp/app/shared/group.ts
import { Member } from "../shared/member"
import { Meeting } from "../shared/meeting"

export class Group {
    groupId: number;
    creationTime: Date;
    meetings: Array<Meeting> = new Array<Meeting>();
    members: Array<GroupMember> = new Array<GroupMember>();
    type: GroupType;
    groupName: string;
}

export enum GroupType {
    Open,
    Closed,
    ByApplication
}

export class GroupMember {
    isAdmin: boolean;
    memberId: string;
    member: Member;
}
